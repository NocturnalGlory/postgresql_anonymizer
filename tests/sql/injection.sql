CREATE EXTENSION IF NOT EXISTS anon CASCADE;

CREATE TABLE a (
    i SERIAL,
    d TIMESTAMP,
    x INTEGER
);

--SELECT anon.random_phone('11; SELECT 0;') LIKE '11; SELECT 0;%'

--SELECT anon.add_noise_on_numeric_column('a; SELECT 1','x',0.5);
--SELECT anon.add_noise_on_numeric_column('a','x; SELECT 1',0.5);

--SELECT anon.add_noise_on_datetime_column('a; SELECT 1','d','2 days');
--SELECT anon.add_noise_on_datetime_column('a','d; SELECT 1','2 days');
--SELECT anon.add_noise_on_datetime_column('a','d','2 days; SELECT 1');

SELECT anon.shuffle_column('a; SELECT 1','x','i');
SELECT anon.shuffle_column('a','x; SELECT 1','i');
SELECT anon.shuffle_column('a','x','i; SELECT 1');

--SELECT anon.load('/dev/null''; CREATE TABLE inject (i int);--');

DROP TABLE a;
DROP EXTENSION anon CASCADE;
