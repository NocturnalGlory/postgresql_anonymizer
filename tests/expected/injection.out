CREATE EXTENSION IF NOT EXISTS anon CASCADE;
-- Sample Table
CREATE TABLE a (
    i SERIAL,
    d TIMESTAMP,
    x INTEGER
);
--
-- random_phone
--
--returns TRUE 
SELECT anon.random_phone('11; SELECT 0;') LIKE '11; SELECT 0;%';
 ?column? 
----------
 t
(1 row)

--
-- add_noise_on_numeric_column
--
-- returns 'invalid name syntax'
SELECT anon.add_noise_on_numeric_column('a; SELECT 1','x',0.5);
ERROR:  invalid name syntax
LINE 1: SELECT anon.add_noise_on_numeric_column('a; SELECT 1','x',0....
                                                ^
-- returns a WARNING and FALSE
SELECT anon.add_noise_on_numeric_column('a','x; SELECT 1',0.5);
WARNING:  Column 'x; SELECT 1' is not present in table 'a'.
 add_noise_on_numeric_column 
-----------------------------
 f
(1 row)

--
-- add_noise_on_datetime_column
--
-- returns 'invalid name syntax'
SELECT anon.add_noise_on_datetime_column('a; SELECT 1','d','2 days');
ERROR:  invalid name syntax
LINE 1: SELECT anon.add_noise_on_datetime_column('a; SELECT 1','d','...
                                                 ^
-- returns a WARNING and FALSE
SELECT anon.add_noise_on_datetime_column('a','d; SELECT 1','2 days');
WARNING:  Column 'd; SELECT 1' is not present in table 'a'.
 add_noise_on_datetime_column 
------------------------------
 f
(1 row)

-- returns 'invalid name syntax'
SELECT anon.add_noise_on_datetime_column('a','d','2 days; SELECT 1');
ERROR:  invalid input syntax for type interval: "2 days; SELECT 1"
LINE 1: SELECT anon.add_noise_on_datetime_column('a','d','2 days; SE...
                                                         ^
--
-- shuffle_column
--
-- returns 'invalid name syntax'
SELECT anon.shuffle_column('a; SELECT 1','x','i');
ERROR:  invalid name syntax
LINE 1: SELECT anon.shuffle_column('a; SELECT 1','x','i');
                                   ^
-- returns a WARNING and FALSE
SELECT anon.shuffle_column('a','x; SELECT 1','i');
WARNING:  Column 'x; SELECT 1' is not present in table 'a'.
 shuffle_column 
----------------
 f
(1 row)

-- returns a WARNING and FALSE
SELECT anon.shuffle_column('a','x','i; SELECT 1');
WARNING:  Column 'i; SELECT 1' is not present in table 'a'.
 shuffle_column 
----------------
 f
(1 row)

--
-- load
--
-- returns a WARNING and FALSE
SELECT anon.load('base/''; CREATE TABLE inject_via_load (i int);--');
WARNING:  The path 'base/'; CREATE TABLE inject_via_load (i int);--' is not correct. Data is not loaded.
 load 
------
 f
(1 row)

SELECT COUNT(*) = 0 
FROM pg_tables
WHERE tablename='inject_via_load';
 ?column? 
----------
 t
(1 row)

--
-- Dynamic Masking
--
-- returns TRUE
SELECT anon.start_dynamic_masking('public','foo; CREATE TABLE inject_via_init (i int);--');
 start_dynamic_masking 
-----------------------
 t
(1 row)

SELECT COUNT(*) = 0 
FROM pg_tables
WHERE tablename='inject_via_init';
 ?column? 
----------
 t
(1 row)

-- CLEAN UP
DROP TABLE a CASCADE;
NOTICE:  drop cascades to view "foo; CREATE TABLE inject_via_init (i int);--".a
DROP EXTENSION anon CASCADE;
NOTICE:  drop cascades to event trigger anon_mask_update
